### PR TITLE
Removing unneeded function calls and empty bodies

### PR DESCRIPTION
**Pull request by pi1024e**
One of the things I found strange when looking at the code is seeing how corerun is literally a main function, with main passing ALL of its parameters to that function.

This is confusing; why not main corerun main?!

So, I did just that.

Please let me know if there are any problems.

Thank you

### CODE REVIEW

**Comment by jkotas** [comment]
build errors

File: src/jit/compiler.cpp [hunk 1]
```diff
@@ -9750,8 +9749,7 @@ int cLeafIR(Compiler* comp, GenTree* tree)
                             switch (tree->GetRegTag())
                             {
                                 case GenTree::GT_REGTAG_REG:
-                                    chars += printf(":%s", comp->compRegVarName(tree->GetRegNum()));
-                                    break;
```
Comment (AaronRobinsonMSFT) [inline src/jit/compiler.cpp]: Fall throughs should be avoided. Please add the `break;` back in all cases.

File: src/jit/compiler.cpp [hunk 2]
```diff
@@ -10692,11 +10685,10 @@ void cNodeIR(Compiler* comp, GenTree* tree)
         chars += printf(" ");
         chars += cLeafIR(comp, tree);
     }
-    else if (op == GT_LEA)
+    else if !(op == GT_LEA) // Already dumped GT_LEA above.
```
Comment (AaronRobinsonMSFT) [inline src/jit/compiler.cpp]: I am not familiar with the `!` outside the parentheses. Is this a new syntax in C++?

Comment (pi1024e) [reply src/jit/compiler.cpp]: I apologize, there were a few commits that should not have slipped in, but did. I fixed this!

**Comment by pi1024e** [comment]
A few commits that were not supposed to slipped through, calling the build errors. I have moved them to the correct branch on my fork and removed them from this PR.

**Comment by jkotas** [comment]
Thank!

**Pull request merged**